docs/notebooks/ask_tell_optimization.pct.py

```diff
@@ -0,0 +1,166 @@
+
+# %%
+model1 = create_model(build_model(initial_data, kernel_func=lambda v: gpflow.kernels.RBF(variance=v)))
+model2 = create_model(build_model(initial_data, kernel_func=lambda v: gpflow.kernels.Matern32(variance=v)))
+
+dataset = initial_data
+for step in range(n_steps):
+    # this criterion is meaningless
+    # but hopefully illustrates the idea!
+    if step % 2 == 0:
+        print("Using model 1")
+        model = model1
+    else:
+        print("Using model 2")
+        model = model2
+
+    print("Asking for new point to observe")
+    ask_tell = AskTellOptimizer(search_space, dataset, model)
+    new_point = ask_tell.ask()
+
+    new_data_point = observer(new_point)
+    dataset = dataset + new_data_point
+
+    print("Training models externally")
+    model1.update(dataset)
+    model1.optimize(dataset)
+
+    model2.update(dataset)
+    model2.optimize(dataset)
+
+plot_ask_tell_regret(ask_tell.to_result())
+
+
+# %% [markdown]
+# ## External experiment: storing optimizer state
+#
+# Now let's suppose you are optimizing a process that takes hours or even days to complete, e.g. a lab experiment or a hyperparameter optimization of a big machine learning model. This time you cannot even express the objective function in Python code. Instead you would like to ask Trieste what configuration to run next, go to the lab, perform the experiment, collect data, feed it back to Trieste and ask for the next configuration, and so on. It would be very convenient to be able to store intermediate optimization state to disk or database or other storage, so that your machine can be switched off while you are waiting for observation results. 
+# 
+# In this section we'll show how you could do it with Ask-Tell in Trieste. Of course we cannot perform a real physical experiment within this notebook, so we will just mimick it ny using pickle to write optimization state and read it back.
+
+# %%
+import pickle
+
+model = build_model(initial_data)
+ask_tell = AskTellOptimizer(search_space, initial_data, model)
+for step in range(n_steps):
+    print(f"Ask Trieste for configuration #{step}")
+    new_config = ask_tell.ask()
+
+    print("Saving Trieste state to re-use later")
+    state: Record[None] = ask_tell.to_record()
+    saved_state = pickle.dumps(state)
+
+    print(f"In the lab running the experiment #{step}.")
+    new_datapoint = scaled_branin(new_config)
+
+    print("Back from the lab")
+    print("Restore optimizer from the saved state")
+    loaded_state = pickle.loads(saved_state)
+    ask_tell = AskTellOptimizer.from_record(loaded_state, search_space)
+    ask_tell.tell(Dataset(new_config, new_datapoint))
+
+
+plot_ask_tell_regret(ask_tell.to_result())
+
+
+# %% [markdown]
+# A word of warning. Serialization is not guaranteed to work smoothly with every GPflow model, so apply caution when using similar technique in your problems.
```

Comment:
perhaps remove GPflow, since we will have multiple model classes?

Reply:
But we can't claim anything about other types of models yet, can we? Although perhaps we can say that this simple serialization technique isn't guaranteed to work with tensorflow in general. Would that sound better?